models/repo/license.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+func init() {
+	db.RegisterModel(new(RepoLicense))
+}
+
+type RepoLicense struct { //revive:disable-line:exported
+	ID          int64 `xorm:"pk autoincr"`
+	RepoID      int64 `xorm:"UNIQUE(s) INDEX NOT NULL"`
+	CommitID    string
```

Comment:
What's the commit_id for? Why not a branch name here?

Reply:
It is used for checking existing licenses when updating licenses:
If the newly detected licenses already exists in DB, we will update the commit ID.
And branch name can not correctly detect it, as same branch name may have different commit id in sometimes?
Maybe both are ok, but commit id is safer I think.